enhancements/worker-latency-profile/worker-latency-profile.md

```diff
@@ -0,0 +1,450 @@
+  // conditions describes the state of the WorkerLatencyProfile and related components
+  // (Kubelet or Controller Manager or Kube API Server)
+  // +patchMergeKey=type
+  // +patchStrategy=merge
+  // +optional
+  Conditions []WorkerLatencyStatusCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+  // relatedObjects is a list of objects that are "interesting" or related to this WorkerLatencyProfile. e.g. KubeletConfig object used for updating Kubelet arguments
+  // +optional
+  RelatedObjects []ObjectReference `json:"relatedObjects,omitempty"`
+}
+
+// WorkerLatencyStatusConditionType is an aspect of WorkerLatencyProfile state.
+type WorkerLatencyStatusConditionType string
+
+const (
+  // Progressing indicates that the updates to component (Kubelet or Controller
+  // Manager or Kube API Server) is actively rolling out, propagating changes to the
+  // respective arguments.
+  WorkerLatencyProfileProgressing WorkerLatencyStatusConditionType = "Progressing"
+
+  // Complete indicates whether the component (Kubelet or Controller Manager or Kube API Server)
+  // is successfully updated the respective arguments.
+  WorkerLatencyProfileComplete WorkerLatencyStatusConditionType = "Complete"
+
+  // Degraded indicates that the component (Kubelet or Controller Manager or Kube API Server)
+  // does not reach the state 'Complete' over a period of time
+  // resulting in either a lower quality or absence of service.
+  // If the component enters in this state, "Default" WorkerLatencyProfileType
+  // rollout will be initiated to restore the respective default arguments of all
+  // components.
+  WorkerLatencyProfileDegraded WorkerLatencyStatusConditionType = "Degraded"
+)
+type ConditionStatus string
+
+// These are valid condition statuses. "ConditionTrue" means a resource is in the condition.
+// "ConditionFalse" means a resource is not in the condition. "ConditionUnknown" means kubernetes
+// can't decide if a resource is in the condition or not.
+const (
+	ConditionTrue    ConditionStatus = "True"
+	ConditionFalse   ConditionStatus = "False"
+	ConditionUnknown ConditionStatus = "Unknown"
+)
+
+type WorkerLatencyStatusConditionOwner string
+
+const (
+  // Machine Config Operator will update condition status by setting this as owner
+  MachineConfigOperator WorkerLatencyStatusConditionOwner = "MachineConfigOperator"
+
+  // Kube Controller Manager Operator will update condition status  by setting this as owner
+  KubeControllerManagerOperator WorkerLatencyStatusConditionOwner = "KubeControllerManagerOperator"
+
+  // Kube API Server Operator will update condition status by setting this as owner
+  KubeAPIServerOperator WorkerLatencyStatusConditionOwner = "KubeAPIServerOperator"
+)
+
+type WorkerLatencyStatusCondition struct {
+	// Owner specifies the operator that is updating this condition
+	// +kubebuilder:validation:Required
+	// +required
+	Owner WorkerLatencyStatusConditionOwner string `json:"owner"`
```

Comment:
IIRC last time we spoke, @deads2k proposed that each operator will report that status inside its own Status, did that change?

Reply:
IIRC, @deads2k mentioned each operator reports the status inside it's own `Status` even today. But I thought that was general statement and wasn't binding to this proposal. 

When the user rolls out `WorkerLatencyProfile`, they need to know the status of that roll out. Any success or failure of an individual component reported in a centralized place will be extremely useful for the user. Such information is also useful for performing a rollback (e.g. if kubelet args fail to get updated, controller manager and API server args can be rolled back by their respective operators by observing this centralized status)